### PR TITLE
[serve] Don't change deployment status when autoscaling

### LINKED ISSUE #35948: [Serve] Introduce new deployment state in the deployment for autoscaling.
### Description

Currently autoscaling event will trigger deployment to be 'UPDATING'. 

It is useful to differentiate the autoscaling state out of the UPDATING state for better observability. (Sometimes autoscaling will trigger too often, it will cause confusion to the user.) 

### Use case

_No response_

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

The deployment state UPDATING should only be used during redeployment. Right now the state is updating during autoscaling, which can be confusing for users. This PR makes it so that the state doesn't change during autoscaling. This usually means that a deployment's status will remain HEALTHY while it's autoscaling.

## Related issue number

Closes https://github.com/ray-project/ray/issues/35948

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by edoakes — APPROVED** [review]

**Review by sihanwang41 — APPROVED** [review]
sync offline, one change needed to always set version (`new_info.version = self._target_state.version.code_version`) . approval for unblocking! LGTM

File: python/ray/serve/_private/deployment_state.py [hunk 1]
```diff
@@ -1289,6 +1289,19 @@ def _set_target_state(self, target_info: DeploymentInfo) -> None:
 
         logger.info(f"Deploying new version of deployment {self._name}.")
 
+    def _set_target_state_autoscaling(self, num_replicas: int) -> None:
+        """Update the target number of replicas based on an autoscaling decision."""
```
[the remark this answers is not in the document]
Comment (zcin) [reply python/ray/serve/_private/deployment_state.py]: Added comments!

File: python/ray/serve/tests/test_autoscaling_policy.py [hunk 2]
```diff
@@ -34,6 +40,18 @@
 import numpy as np
 
 
+def wait_for_condition_raise(
+    condition_predictor, timeout=10, retry_interval_ms=100, **kwargs: Any
+):
+    """Wait until a condition is met. If exception occurs, raise it."""
```
[the remark this answers is not in the document]
Comment (zcin) [reply python/ray/serve/tests/test_autoscaling_policy.py]: Good point, I've added this as a parameter to wait_for_condition

**Comment by zcin** [comment]
Thanks @sihanwang41, I've addressed your comment!

@edoakes Tests are passing, should be ready to merge!

**Pull request merged**